app/controllers/users_controller.rb

```diff
@@ -0,0 +1,64 @@
+class UsersController < ApplicationController
+  # this is a filter to restrict access only to logged in users:
+  # NOTE that we apply ONLY so that this check/filter only occurs on the edit and :edit and :update actions
+  before_action :logged_in_user, only: [:edit, :update]
```

Comment:
before_action :require_log_in
 before_action :authorize_user

and comments won't be necessary, method names should be clear enough to tell you what is going on.